torch/csrc/jit/script/tree.h

```diff
@@ -115,30 +112,24 @@ struct String : public Tree {
   std::string value_;
 };
 struct Number : public Tree {
-  Number(double value_) : Tree(TK_NUMBER), value_(value_) {}
-  virtual double doubleValue() const override {
+  Number(SourceRange range, std::string value)
+    : Tree(TK_NUMBER)
+    , range_(std::move(range))
+    , value_(std::move(value)) {}
+  virtual const std::string& stringValue() const override {
     return value_;
   }
-  template <typename... Args>
-  static TreeRef create(Args&&... args) {
-    return std::make_shared<Number>(std::forward<Args>(args)...);
-  }
-
- private:
-  double value_;
-};
-struct Bool : public Tree {
-  Bool(bool value_) : Tree(TK_BOOL), value_(value_) {}
-  virtual double doubleValue() const override {
-    return value_;
+  virtual const SourceRange& range() const override {
```

Comment:
Previously only CompoundTrees have ranges. The expectation is that anything that requires a range will be wrapped in another Tree, e.g.:

`(const 4)`
Where the 4 is a String tree.